linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -1799,11 +1801,16 @@ private boolean innerHasNext() {
             i = -1;
           }
 
-          @Override public void close() {
-            outerEnumerator.close();
+          private void closeInner() {
```

Comment:
Should we use `synchronized`?

Reply:
Interesting point. 
Maybe I'm wrong, but my understanding is that our enumerators are not thread-safe, so if a 3rd party uses them on a multi-threaded environment, they would need to handle synchronization on their end.

Reply:
Get it